### ISSUE TITLE
Should we deprecate get_sun

### ISSUE THREAD

**Issue by StuartLittlefair** [astropy/astropy#14368]
### What is the problem this feature will solve?

In #14354 we removed `get_moon` since it was nothing but a wrapper around `get_body`.

The case of `get_sun` is more complex; it provides an alternative method to `get_body(‘sun’)`. Both (by default) use `erfa.epv00` to find the relative position of Sun and Earth, but they differ in their handling of light travel time and aberration.

Arguably one is more correct than the other, and we should only retain the correct implementation? However in https://github.com/astropy/astropy/pull/14354#issuecomment-1419656506, @eerovaher pointed out that @eteq was strongly in favour of retaining `get_sun` in the past, so perhaps this needs more careful discussion?

### Describe the desired outcome

Only one way is provided of getting the Sun’s position

**Comment by mkbrewer** [comment]
While `get_sun` ignores light travel time, that is likely insignificant anyway as the Sun doesn't move very far in 8 minutes. It seems to me that its more important deficiency is that it is purely geocentric.  and therefore of limited use.

**Comment by mhvk** [comment]
Agreed with @mkbrewer - @StuartLittlefair, maybe the best way forward is simply to start deprecating it?

**Comment by alexandrecesarba** [comment]
Hello Astropy community,

I hope everyone is doing well. After a thorough review and the associated discussions, I'm considering submitting a PR to deprecate the `get_sun` function. Here are my reasons:

1. **Consistency and Clarity**: The `get_sun` function serves a similar purpose to `get_body('sun')`. Having two functions with overlapping functionalities can be confusing for users, especially newcomers. Deprecating `get_sun` would streamline the API and make it more intuitive.

2. **Technical Considerations**: While `get_sun` ignores light travel time, which might be deemed insignificant given the Sun's minimal movement in 8 minutes, its purely geocentric nature limits its broader applicability. On the other hand, `get_body('sun')` offers a more comprehensive approach, accounting for both light travel time and aberration.

3. **Maintenance Overhead**: Maintaining two functions that serve similar purposes can lead to duplicated efforts in terms of bug fixes, enhancements, and documentation updates. By focusing our efforts on a single function, we can ensure more efficient maintenance and consistent improvements.

4. **Guided Transition**: To ensure a smooth transition for users, I propose adding clear deprecation warnings to `get_sun`, guiding users towards `get_body('sun')`. Additionally, we can enhance the documentation to provide clear migration paths and examples.

I understand the historical significance and the reasons `get_sun` was favored in the past. However, as the library evolves, it's essential to make decisions that benefit the broader user base and the maintainability of the codebase.

I'd love to hear your thoughts, feedback, and any concerns you might have regarding this proposal. Your insights will be invaluable in ensuring that we make the best decision for the Astropy community.

Thank you for your time and consideration.

Best regards,
Alexandre

**Comment by eerovaher** [comment]
I'm still interested to learn why @eteq [thought `get_sun()` should __not__ be replaced](https://github.com/astropy/astropy/pull/4952#issuecomment-220379031) when this was discussed in #4952.

**Comment by eteq** [comment]
(I will try to carve out time to satisfy myself of those two things, but someone else should feel free to try it and report the results back if they can get there first!)

**Comment by mkbrewer** [comment]
They aren't going to be the same to machine precision due to the compensation for light travel time in `get_body('sun')`. This will be small, but I doubt that it is machine precision small. Also, `get_body('sun')` is bound to be slower due to the light travel time calculation plus the fact that the Earth's barycentric and heliocentric position needs to be calculated twice in order to first calculate the barycentric position of the Sun and then add the barycentric position of the Earth in the transormation to GCRS..

**Comment by mkbrewer** [comment]
The advantages of using `get_body('sun')` are that the position of the Sun is calculated in a manner consistent with that of the Moon and planets and it offers the ability to calculate the topocentric position of the Sun in addition to the geocentric position returned by `get_sun()`.
